common/utils/localStorage.ts

```diff
@@ -62,3 +66,40 @@ async function getWalletConfigKey(wallet: IWallet): Promise<string> {
   const address = await wallet.getAddressString();
   return sha256(`${address}-mycrypto`).toString('hex');
 }
+
+export interface SavedTransaction {
+  hash: string;
+  to: string;
+  from: string;
+  value: string;
+  chainId: number;
+  time: number;
+}
+
+export function saveRecentTransaction(hash: string, tx: EthTx) {
+  try {
+    const fields = getTransactionFields(tx);
+    const from = hexEncodeData(tx.getSenderAddress());
+    const txObj: SavedTransaction = {
+      hash,
+      to: toChecksumAddress(fields.to),
+      from: toChecksumAddress(from),
+      value: fields.value,
+      chainId: fields.chainId,
+      time: Date.now()
+    };
+    const recentTxs = [txObj, ...loadRecentTransactions()];
+    localStorage.setItem('recent-transactions', JSON.stringify(recentTxs));
```

Comment:
Can this be made into an enum instead?

Reply:
On second thought, this should probably just be stored within the redux state itself under `recent`, avoiding the use of separate keys and binding the data to the redux state